moveit_ros/planning/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp

```diff
@@ -303,36 +346,100 @@ moveit::core::SolverAllocatorFn KinematicsPluginLoader::getLoaderFunction(const
 
       if (default_solver_plugin_.empty())
       {
-        RCLCPP_DEBUG(LOGGER, "Loading settings for kinematics solvers from the ROS param server ...");
+        // construct a parameter client to get parameters from the move_group node
```

Comment:
Wouldn't it make sense to move this into step 3?

Reply:
This was to avoid creating it in the loop.